### PR TITLE
AWS S3 Setup for Drupal

### PR DESCRIPTION
**Pull request by rachelwhitton**
Replaces #1870 

## Effect
PR includes the following changes:
- New doc: AWS S3 Setup for Drupal

@peter-pantheon we'll go forward with this PR in favor of the one submitted from your fork - this way all contributors have write access for easier collaboration. 

I made quite a few edits in a first pass, mind taking a look at the changes? 

cc @nataliejeremy

### CODE REVIEW

File: source/_docs/drupal-s3.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  <div role="tabpanel" class="tab-pane" id="d8s3fs">
+   <br/>
+   <p>Install the <a href="https://www.drupal.org/project/s3fs">S3 File System</a> module and AWS SDK version 3.x library using Composer.</p>   
+   First, ensure that the Composer for your site will first look to Drupal's preferred package source to find modules:
+   <pre><code class="bash hljs">
+   composer config repositories.drupal composer https://packages.drupal.org/8
+   </code></pre>
+   Install s3fs module from the preferred package source:
+   <pre><code class="bash hljs">
+   composer require drupal/s3fs --prefer-dist
```
Comment (rachelwhitton) [inline source/_docs/drupal-s3.md]: @peter-pantheon Do users need to set installation paths or handle the vendor dir a certain way? What should be committed to the code repository?

File: source/_docs/drupal-s3.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  <div role="tabpanel" class="tab-pane active" id="d7s3fs">
+  <br/>
+  Install the <a href="https://www.drupal.org/project/libraries">Libraries API</a> and <a href="https://www.drupal.org/project/s3fs">S3 File System</a> modules:
+  <pre><code class="bash hljs">
+  terminus drush 'en libraries s3fs -y'
+  </code></pre>
+
+  Get the <a href="https://github.com/aws/aws-sdk-php/releases">AWS SDK Library 2.x</a>:
+  <pre><code class="bash hljs">
+  terminus drush 'make --no-core sites/all/modules/contrib/s3fs/s3fs.make'
```
Comment (rachelwhitton) [inline source/_docs/drupal-s3.md]: I wasn't able to run this successfully, any tips @peter-pantheon? 

```
[2016-09-12 21:02:30] [info] Running drush make --no-core sites/all/modules/contrib/s3fs/s3fs.make -y on drupal-7-aws-test-dev
    cmd: 'make --no-core sites/all/modules/contrib/s3fs/s3fs.make -y'
    site: 'drupal-7-aws-test'
    env: 'dev'
file_get_contents(sites/all/modules/contrib/s3fs/s3fs.make): failed    [warning]
to open stream: No such file or directory make.utilities.inc:577
Invalid or empty make file: sites/all/modules/contrib/s3fs/s3fs.make     [error]
Beginning to build sites/all/modules/contrib/s3fs/s3fs.make.                [ok]
Make new site in the current directory? (y/n): y
```

File: source/_docs/drupal-s3.md (hunk 1 above)
Comment (peter-pantheon) [inline source/_docs/drupal-s3.md]: @rachelwhitton In D8 the standard installation of D8 modules and their required libraries via Composer should need no additional special configuration. How to handle the vendor directory is actually up to the user and their workflow pattern, see: [Decide Whether to Commit the Vendor Directory|https://pantheon.io/docs/composer-drupal-8/#decide-whether-to-commit-the-vendor-directory].

File: source/_docs/drupal-s3.md (hunk 2 above)
Comment (peter-pantheon) [inline source/_docs/drupal-s3.md]: @rachelwhitton Probably could use your expertise on how to add this additional info. The make file expects the s3fs module to be in sites/all/modules/contrib and that is not specified in the step-by-step: that before running the drush commands to install the modules one should ensure that there are contrib and custom subdirs in sites/all/modules/.

File: source/_docs/drupal-s3.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+</ul>
+<!-- Tab panes -->
+<div class="tab-content">
+ <div role="tabpanel" class="tab-pane active" id="d7media">
+ <br/>
+ To have solid, viewable, manageable Drupal 7 file handling (See: <a href="https://www.drupal.org/node/1699054">Media 2.x Quick Start Guide</a>), install:<br />   
+ Ctools, File Entity, Views, and Entity View Modes
+ <pre><code class="bash hljs">terminus drush 'en ctools file_entity views entity_view_mode -y'</code></pre>
+
+ Then install the Media module 2.x (e.g. 7.x-2.x-dev)
```
Comment (rachelwhitton) [inline source/_docs/drupal-s3.md]: @peter-pantheon just noticed [**Media: Filesystem**](https://pantheon.io/docs/unsupported-modules-plugins/#media-filesystem) is listed as a problematic module - is this the same one?

Comment (peter-pantheon) [inline source/_docs/drupal-s3.md]: Yes. It is. Missed that. This whole section on installing Media module is actually extraneous. It is my own prior  experience with Media module that leads me to suggest this. This should be removed to keep this as clean and simple as possible.

**Pull request merged**